lib/posix.lua.in

```diff
@@ -519,5 +519,27 @@ if _DEBUG ~= false then
   end
 end
 
+local function prepare_args (pat_mark)
+  local pat, mark = "*", 0
+  if type (pat_mark) == "table" then
+    if type (pat_mark.pattern) == "string" then
+      pat = pat_mark.pattern
+    end
+    if pat_mark.MARK then
+      mark = M.GLOB_MARK
+    end
+  elseif type (pat_mark) == "string" then
+    pat = pat_mark
+  end
+
+  return pat, mark
+end
+
+local cglob = M.glob
```

Comment:
I would remove the import of `posix.glob` imports into `M` right at the top of the file, and then no need to cache that here... we can just refer to `posix.glob.glob` and `posix.glob.GLOB_MARK` directly when we need them.